### PR TITLE
 [SPARK-11976][SPARKR] Support "." character in DataFrame column name

### PR DESCRIPTION
**Pull request by rerngvit**
```
## What changes were proposed in this pull request?
- Add support "." character in DataFrame column name
- Remove R code in "createDataFrame()" that replaces "." with "_"
- Remove warning suppression in createDataFrame() for iris dataset (introduced in SPARK-12034)

## How was this patch tested?
SparkR unit tests and manual testing with the R script described in SPARK-11976
```

### CODE REVIEW

**Comment by rerngvit** [comment]
@sun-rui Please have a look.

**Comment by shivaram** [comment]
Jenkins, ok to test

**Comment by felixcheung** [comment]
There's this (and I thought one more, if I recall):
https://github.com/apache/spark/blob/master/R/pkg/R/DataFrame.R#L349

And its tests in test_sparkSQL.R L779 - I'm surprised these tests pass?

**Comment by sun-rui** [comment]
@rerngvit, could you share the background that this PR can fix the issue. I see that https://issues.apache.org/jira/browse/SPARK-11976 is still open. Any other PR in Spark 2.0 make this possible?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala [hunk 1]
```diff
@@ -201,6 +201,8 @@ abstract class LogicalPlan extends QueryPlan[LogicalPlan] with Logging {
       attribute: Attribute): Option[(Attribute, List[String])] = {
     if (!attribute.isGenerated && resolver(attribute.name, nameParts.head)) {
       Option((attribute.withName(nameParts.head), nameParts.tail.toList))
+    } else if (!attribute.isGenerated && resolver(attribute.name, nameParts.mkString("."))) {
+      Option((attribute.withName(nameParts.mkString(".")), Nil))
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala]: Hi, I'm just curious. Is it okay for other Spark module?
> Different from resolveAsTableColumn, this assumes `name` does NOT start with a qualifier.

**Comment by rerngvit** [comment]
@felixcheung 
I removed prohibition for "." in the function "colnames()" for SparkR in the recent patch and also its test in the file you mentioned (test_sparkSQL.R L779). Feel free to have a look.

**Comment by rerngvit** [comment]
@felixcheung: in the recent patch, I removed checking for "." for function colnames() and its test code for the file you indicated.

**Comment by rerngvit** [comment]
@sun-rui
>  could you share the background that this PR can fix the issue.
As stated in https://issues.apache.org/jira/browse/SPARK-11976, Spark core is already supporting column name ".".  However, it does not work without the backticks (`). The missing part to make it work is a corresponding logic in the Analyzer layer.

>  I see that https://issues.apache.org/jira/browse/SPARK-11976 is still open.
I do not fully understand your question. Could you please elaborate a bit more? **This PR** is actually SPARK-11976.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala (hunk 1 above)
Comment (rerngvit) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala]: @dongjoon-hyun I do not fully understand your question. Could you please elaborate a bit more? What do you mean by "okay for other Spark module"?

**Comment by sun-rui** [comment]
@rerngvit, sorry, I mean https://issues.apache.org/jira/browse/SPARK-11977. If your PR can enable accesses to columns with "." in their names without backticks, please first submit a PR for SPARK-11977, as the change is for the Spark Core, not SparkR specific. After that PR gets merged, you can then submit a PR for SPARK-11976 which contains SparkR only changes.

**Comment by rerngvit** [comment]
@sun-rui Thanks. I understand now. I would submit a PR to SPARK-11977 first.

**Comment by sun-rui** [comment]
@rerngvit, I modifed the title of SPARK-11977 to a narrow scope. You can go for it.

**Comment by rerngvit** [comment]
Since SPARK-11977 didn't get merge and this PR is blocked on that. I decided to close this PR.

**Pull request closed without merging**